### DISCUSSION TITLE
(Vector) Hide line tool

### DISCUSSION THREAD

**Discussion by ghost** [opentoonz/opentoonz#6464] [Ideas]
<details>Hi folks,

Currently to hide specifics parts of drawing on OT we have to cut the parts, then use 0 thickness or 0 alpha. But this process is long and break the tweening process. 

I would like in the future to see a feature to hide the strokes in opentoonz, something intuitive and above all, very fast without damaging the existing tweening process.

I think we should think about the best integration for Opentoonz with its current tools, all ideas will be welcome, there is already a discussion link on the forum https://groups.google.com/forum/#!category-topic/opentoonz_en/feature-requests/dsME50E4t8g  and https://groups.google.com/forum/#!category-topic/opentoonz_en/feature-requests/G_AgxoI37QY  and here  https://github.com/opentoonz/opentoonz/issues/746 on Github.

I would like to submit a bountysource because it is a very useful feature in many situations without having to use masks, cutter or erasers : Bodyturns, headturns, parts of body, cycles, cut-out etc ...

Personally, I like 2 softwares approaches. Moho and Cacani. I much prefer the cacani approach because it allows maximum flexibility and customization that also includes the moho approach, but the simple moho approach will also save significant time anyway compared to the workflow currently available in OT. 

Here is my proposal on this topic:

Creation / integration of a "hide tool" with the properties of some tools currently available in OT, essentially the selection tool.

- On the one hand :
we could choose the way we want to hide our stroke as Type : endpoint to endpoint (segment) / Freehand / polyline/ Freely , the last type should allows maximum flexibility , in this way we can choose that we want to hide easly. Cf. See video and picture below.

![hide tool request](https://user-images.githubusercontent.com/29790512/32406322-b67b2960-c176-11e7-8e1d-e4b2997099a6.gif)

![hide tool](https://user-images.githubusercontent.com/29790512/32406335-dc099874-c176-11e7-85a2-4865bf87f2fa.png)

- And on the other hand :
the mode of application found in the selection tool to be able to quickly affect specific strokes as shown above.

Video above it's how Cacani works on this :
2 segments tools  one to use by segment and the other freely on the parts of the drawing to hide. 
That said, there are others approachs about this feature request, the main thing would be to provide Opentoonz with an intuitive tool in the future to easily hide lines without breaking them.

@ developers : Currently Bountysource don't works with github account, I will submit that once it will be repaired.

</details>

Updated this request. 

I hidden the original post because with the addition of the segment eraser in OT, I think it's more relevant to base most of this request on its behavior or approach, although that remains open.

The tool for hiding strokes would work very similarly to the segment eraser tool but keeping the strokes

- invisible
- without breaking them at the intersection

in order to make the auto inbetween process always correct.

Cacani which has tools for hiding lines, similar to the segment eraser tool, works this way with its segment tool.

Unlike the eraser, the line is not destroyed but just hidden. This makes the cleaning process just as clear as with the eraser and with the benefits of keeping the auto inbetween functional for hidden lines.

![Cacani segment tool](https://user-images.githubusercontent.com/29790512/103144749-eae6e400-472e-11eb-8f03-658c4d84346a.gif)

There is no bounty currently but this feature remains open to the initial bounty, if interested for its implementation.

**Comment by jpturcotte** [comment]
@openanim Nice initiative! There is a partial workaround for logging in to Bountysource here: https://github.com/bountysource/core/issues/1147#issuecomment-337741285.

**Comment by ghost** [comment]
@jpturcotte Thanks for the tip, I will try it !

**Comment by ghost** [comment]
I created the bounty here :

https://www.bountysource.com/issues/51160151-hide-tool-request

**Comment by Orphanlast** [comment]
So, I'm confused. Is this like a layer mask (as seen on photoshop) because that would legitimately interest me. 

But what you're requesting. Are you saying you want to hide lines that are hidden in the edit mode but those lines will still render and be viewable in the preview mode.

Forgive me, I only know how Opentoonz works. It's the only animation software I understand. So sometimes when someone says "I want this to work like THIS program," I just have no idea what they're talking about some times.

This legitimately interests me, and you may be able to get some donations with my "Opentoonz News" videos. They're a good form of community outreach.

Furthermore, if I see this bounty gets consistent donations, I'll report on the bounty multiple times until it finally reaches the Bounty Hunter's goal.

Please answer these questions. If this is news worthy, I'll report it in the News.

**Comment by ghost** [comment]
In fact, it is not a "mask "but can be used instead of it in some cases. The advantage is that it is very fast and very flexible. As mentioned above, there are many situations where this is useful, in a vector process workflow. I invite you to read the previous request about that here  #746 to see some exemples.

Currently OT can already hide entire lines. It should be allow to hide segments (intersections of lines) too, without having to use the cutter tool or others destructive tools beforehand. It misses this aspect which would allow more speed / flexibility. If the tool pump or selection could be used on intersection of lines, or even better if it can be used totally freely, it would be the icing on the cake xD.

**Comment by ghost** [comment]
Another option, would be to take advantage of the eraser tool but instead of destroying the strokes, this new second eraser or option would make invisible the erased area. Sure, I assume that writing of code would not be obvious but it is perhaps the best solution to this request that would solve everything mentioned here with the parameters already available of vector eraser. I really think that such a feature would be huge for opentoonz.

**Comment by RodneyBaker** [comment]
This feature WIP remains open due to bounty posted but is slipping to 2019 development cycle as no one has yet expressed interest in implementation.

If there is a programmer interested but the bounty is not sufficient please feel free to suggestion an adequate budget.

**Comment by ghost** [comment]
Indeed, I suppose it is better that a developer express his interest himself. i think bountysource should add a tag in the other direction too. Closing for now.

**Comment by RodneyBaker** [comment]
I'll also add...
I was looking at the source code and the areas I saw relating to this issue look inviting enough that I hope this can be pursued by some of us that aren't programmers under the watchful eye of seasoned programmers in an effort to learn how to better assist in implementing fixes like this.

We might start for instance with examining what the code currently does and then make a determination on how best to modify it to achieve the hidden line segments.

We aren't likely to get this right straight out of the gate... but we'll learn a lot in the process.  :)

**Comment by RodneyBaker** [comment]
I'm not sure what triggers the update of the automatic bounty posting here in github.
The link suggests there is a $90 bounty when currently there is none.
Perhaps it takes a day or two to update?

Update: Leaving the bounty tag for now as this proposal did have one previously.
As such, those interested in implementation are encouraged to consider posting a bounty.

**Comment by RodneyBaker** [comment]
Removed WIP from title and bounty label as there is currently no active bounty related to this feature request.

The roadmap label remains because most of the code is present for this feature already and general workflow and tooling can be seen in both the Segment Erase Tool as well as Line Fill (for toonz raster levels).  The latter demonstrates a working equivalent to the hide line tool in many respects.

Update:  Removed 'roadmap' as there is no developer attached and no clear path to the solution.

Recommend documenting the desired solution to utmost degree to better assist in the development of the proposed feature.  Thanks to feedback here, much of that documentation appears in this report already.

**Comment by RodneyBaker** [comment]
Adjusting this request a little as the ability to hide segments exists with Fill Tool for Toonz Raster levels.  While there is Erase Tool Segment mode that will remove/delete vector lines it doesn't have a similar option to hide those lines instead of deleting them.

The most obvious addition would be to add an option that instead of Deleting segments when using Segement Erase the tagged segments would be assigned a transparent style.  A challenge in this is that we would not (necessarily) want to create a new style for each set of segments tagged.

Bottom line:  It would be useful to compare the Fill Tool's segmenting code to see how the part for Toonz raster is being used and how it might transfer to Segement Erase to store the segments in  a transparent style.

Workarounds for the workflow that requires this capability abound.
If working in Toonz Raster this capabiolity to hide lines exists via the Fill Tool.
